library/src/main/java/io/ona/kujaku/KujakuLibrary.java

```diff
@@ -0,0 +1,109 @@
+package io.ona.kujaku;
+
+import android.content.Intent;
+import android.os.AsyncTask;
+import android.util.Log;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.util.List;
+
+import io.ona.kujaku.activities.MapActivity;
+import io.ona.kujaku.callables.AsyncTaskCallable;
+import io.ona.kujaku.domain.Point;
+import io.ona.kujaku.listeners.OnFinishedListener;
+import io.ona.kujaku.tasks.GenericAsyncTask;
+import io.ona.kujaku.utils.Constants;
+
+import static io.ona.kujaku.utils.IOUtil.readInputStreamAsString;
+
+/**
+ * @author Vincent Karuri
+ */
+public class KujakuLibrary {
+
+    private BaseKujakuApplication hostApplication;
+    private static KujakuLibrary library;
+    private List<Point> mapActivityPoints;
+    private static final String TAG = KujakuLibrary.class.getName();
+
+    private KujakuLibrary() {}
+
+    public static KujakuLibrary getInstance() {
+        if (library == null) {
+            library = new KujakuLibrary();
+        }
+        return library;
+    }
+
+    public void sendFeatureJSONToHostApp(JSONObject featureJSON) {
+        getHostApplication().processFeatureJSON(featureJSON);
+    }
+
+    public void sendFeatureJSONToGeoWidget(JSONObject featureJSON) {
+        // TODO: implement this
+    }
+
+    public void setHostApplication(BaseKujakuApplication hostApplication) { this.hostApplication = hostApplication; }
+
+    public BaseKujakuApplication getHostApplication() {
+        return hostApplication;
+    }
+
+
+    public void setMapActivityPoints(List<Point> points) {
+       mapActivityPoints = points;
+    }
+
+    public List<Point> getMapActivityPoints() {
+        return mapActivityPoints;
+    }
+
+    public void launchMapActivity(List<Point> points) {
+        setMapActivityPoints(points);
+
+        Intent intent = new Intent(getHostApplication(), MapActivity.class);
+        createFinalStyleUsingSavedPoints(new OnFinishedListener() {
+            @Override
+            public void onSuccess(Object[] objects) {
+                JSONObject mapboxStyleJSON = (JSONObject) objects[0];
+                if (mapboxStyleJSON != null) {
+                    intent.putExtra(Constants.PARCELABLE_KEY_MAPBOX_STYLES, new String[]{
+                            mapboxStyleJSON.toString()
+                    });
+                    intent.putExtra(Constants.PARCELABLE_KEY_MAPBOX_ACCESS_TOKEN, BuildConfig.MAPBOX_SDK_ACCESS_TOKEN);
+                    intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+                    getHostApplication().startActivity(intent);
+                }
+            }
+            @Override
+            public void onError(Exception e) {
+                Log.e(TAG, Log.getStackTraceString(e));
+            }
+        });
+    }
+
+    private void createFinalStyleUsingSavedPoints(OnFinishedListener onFinishedListener) {
```

Comment:
I preserved this code from an earlier implementation.

Maybe you could share what it was intended to do?

Reply:
If it doesn't serve any purpose, could this method be removed or renamed.

Reply:
I believe it has to do with adding an mspray style to the map activity. I have visually tested the appearance with and without the styling.

I was trying to find out the purpose from you before altering, since you wrote the method.

I think we should keep it as an illustration of custom layering.

I have also changed the method name.